monitor/kafka.go

```diff
@@ -0,0 +1,126 @@
+package monitor
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"github.com/segmentio/kafka-go"
+	"github.com/segmentio/kafka-go/sasl/plain"
+)
+
+const (
+	KafkaBatchInterval  = 1 * time.Second
+	KafkaRequestTimeout = 60 * time.Second
+)
+
+type KafkaProducer struct {
+	writer *kafka.Writer
+	topic  string
+}
+
+type GatewayEvent struct {
+	ID        *string     `json:"id,omitempty"`
+	Type      *string     `json:"type"`
+	Timestamp *string     `json:"timestamp"`
+	Gateway   *string     `json:"gateway,omitempty"`
+	Data      interface{} `json:"data"`
+}
+
+var kafkaProducer *KafkaProducer
+
+func InitKafkaProducer(bootstrapServers, user, password, topic string) error {
+	producer, err := NewKafkaProducer(bootstrapServers, user, password, topic)
+	if err != nil {
+		return err
+	}
+	kafkaProducer = producer
+	return nil
+}
+
+func NewKafkaProducer(bootstrapServers, user, password, topic string) (*KafkaProducer, error) {
+	dialer := &kafka.Dialer{
+		Timeout: KafkaRequestTimeout,
+		SASLMechanism: plain.Mechanism{
+			Username: user,
+			Password: password,
+		},
+		DualStack: true,
+		TLS: &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		},
+	}
+
+	// Create a new Kafka writer
+	writer := kafka.NewWriter(kafka.WriterConfig{
+		Brokers:  []string{bootstrapServers},
+		Topic:    topic,
+		Balancer: kafka.CRC32Balancer{},
+		Dialer:   dialer,
+	})
+
+	return &KafkaProducer{
+		writer: writer,
+		topic:  topic,
+	}, nil
+}
+
+func (p *KafkaProducer) SendEvent(event GatewayEvent) {
+	p.SendEvents([]GatewayEvent{event})
+}
+
+func (p *KafkaProducer) SendEvents(events []GatewayEvent) {
```

Comment:
Is it possible to make it package-private?

Reply:
Done